tests/integration/security/authorization_test.go

```diff
@@ -455,24 +457,37 @@ func TestAuthorization_NegativeMatch(t *testing.T) {
 				"vm":         vm[0].Config().Service,
 			}, "testdata/authz/v1beta1-negative-match.yaml.tmpl").ApplyOrFail(t)
 
+			type request struct {
+				method string // default value is GET if not specified
+				host   string // only set if not empty
+				port   string // default value is http if not specified
+			}
+
 			for _, srcCluster := range t.Clusters() {
+				reqParam := request{method: "GET", port: "http"}
+				reqNotMethodParam := request{method: "PUT", port: "http"}
+				reqNotPortParam := request{method: "GET", port: "http-8091"}
+				reqNotHostParam := request{method: "GET", port: "http", host: "deny.com"}
+
 				a := match.And(match.Cluster(srcCluster), match.Namespace(apps.Namespace1.Name())).GetMatches(apps.A)
 				bInNS2 := match.And(match.Cluster(srcCluster), match.Namespace(apps.Namespace2.Name())).GetMatches(apps.B)
 				if len(a) == 0 || len(bInNS2) == 0 {
 					continue
 				}
 
 				t.NewSubTestf("From %s", srcCluster.StableName()).Run(func(t framework.TestContext) {
-					newTestCase := func(from echo.Instance, to echo.Target, path string, expectAllowed bool) func(t framework.TestContext) {
+					newTestCase := func(from echo.Instance, to echo.Target, path string, expectAllowed bool, request request) func(t framework.TestContext) {
```

Comment:
you can create another wrapper function: `newTestCaseWithRequest(from echo.Instance, to echo.Target, path string expectAllowed bool, req *request)` and then change `newTestCase` to call it with the `reqParam` so that you can just pass the `reqParam` variable once and do not need to repeat it everywhere.